### ISSUE TITLE
Wrong limits for zstd compression level

### ISSUE THREAD

**Issue by gkleen** [pgbackrest/pgbackrest#1931]
1. pgBackRest version:

    2.41
    
2. PostgreSQL version:

    [14.6](https://github.com/NixOS/nixpkgs/blob/dcede1059bc1d5f07961d9d8378158be3fc37ebd/pkgs/servers/sql/postgresql/default.nix#L233)

5. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:

    nixos-unstable

7. Did you install pgBackRest from source or from a package?

    from a [package](https://github.com/NixOS/nixpkgs/blob/d72eb969b070d8b3e6bfaba908577aa7656d197b/pkgs/tools/backup/pgbackrest/default.nix)

8. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    
        ```
        [global]
        …
        compress-level=10
        compress-type=zst
        
        …
        ```
    - ~~`postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)~~
    - errors in the postgresql log file before or during the time you experienced the issue
    
        `ERROR: [032]: '10' is out of range for 'compress-level' option`
    - ~~log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)~~

9. Describe the issue:
    
    AFAIK zstd compression levels go to 22

**Comment by jreidthompson** [comment]
There is a project board card in place regarding this. https://github.com/pgbackrest/pgbackrest/projects/2#card-57714899

**Comment by dwsteele** [comment]
We do need to fix this, but to be honest it is not a high priority. I'm doubtful that these high levels are very useful, but of course it makes sense to support them for completeness if nothing else.

**Comment by dwsteele** [comment]
Fixed in PR #1962. @gkleen How would you like your name to appear in the release notes?

**Comment by gkleen** [comment]
> Fixed in PR #1962. @gkleen How would you like your name to appear in the release notes?

gkleen is fine.
